src/lighteval/main_baseline.py

```diff
@@ -0,0 +1,88 @@
+# MIT License
+
+# Copyright (c) 2024 The HuggingFace Team
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from lighteval.logging.evaluation_tracker import EvaluationTracker
+from lighteval.metrics.utils.metric_utils import MetricCategory
+from lighteval.models.abstract_model import ModelInfo
+from lighteval.tasks.lighteval_task import LightevalTask
+from lighteval.tasks.registry import Registry, taskinfo_selector
+from lighteval.utils.utils import as_list
+
+
+def main(args):
+    """
+    Compute baselines for given tasks.
+
+    It has been tested with generative and accuracy tasks, but may not work correctly for other task types.
+
+    The baseline is computed as follows:
+    - For multiple-choice tasks: It assumes random guessing, so the score is n_correct/number_of_choices.
+    - For other metrics: It assigns a score of 0, which may not be appropriate for all task types.
+
+    Note:
+        This baseline computation may not be suitable for all task types and should be used with caution.
```

Comment:
It would be way better to update the dummy than add a new main script which will be confusing for users

Reply:
Let me go through the code again tmr morning, something seemed obvious an hour ago

Reply:
Went through things again, I agree, my ideas are way more costly than this. 
We should remove the dummy model then as it will always be more costly too. 
Wdyt @NathanHB ?

Reply:
I'm fine with removing the dummy model, it was only used for fast debugging on task releated issues. Does this baseline implem allow for debugging the logging of tasks ?

Reply:
I would keep the dummy model, it's useful when someone just need to passthrough the pipeline workflow. @NathanHB What do you mean by logging of tasks ?

Reply:
I meant loading, and I agree it's needed for quick debugging, we should keep it then.